### PR TITLE
Fix #3571 and made scrolling in windowed mode more generic

### LINKED ISSUE #3571: Can't scroll to right with some scale factors
Repost from https://openrct2.org/forums/topic/1274-cant-scroll-view-to-the-right-when-using-window-scale-factor-of-150/#comment-6586

> Hi, the issue is like the title says. I like to use 1600x900 resolution (native) with a window scale factor of 1.50. I know it takes away some of the detail but I'm used to that zoom, since I played RCT1 for years in a 640x480 screen. Also since in RCT2 you can zoom-out 3x I think it's good to have a closer look because the most zoomed-out will still show the full map.
>
> Anyway... When I hover the mouse over the right side it won't move the screen, which is annoying because I have to right click and I accidentaly delete stuff while building if I try to move around with right-click.  At 1.25 scale factor it works normally. I know it probably has something to do with the fact that it's just zoomed-in for me with the UI adjusted, but in fact my in-game screen still has stuff to the right. But is there any fix to this? Something that could be implemented?
>
> Edit: I'm using the most recent developer build, but the issue was around for me way before. 

**TL;DR:** With some resolution and scale factor combinations (e.g. 1600x900, 1.5), it is impossible to trigger map scrolling when moving cursor to the right edge of window.

### PR DESCRIPTION
**Pull request by marcovmun**
Fix bug #3571
And now in windowed mode when you go out of window the map scroll always in when Scroll view when pointer at screen edge is enabled.
In the old situation that only happens left and above the window

### CODE REVIEW

**Comment by mrtnptrs** [comment]
Sorry to annoy you, but isn't this ready for merge? And why did the appveyor check fail?

**Comment by Nubbie** [comment]
Appveyor sometimes fail, this can't possibly have triggered it, was something else

**Comment by Gymnasiast** [comment]
Tested it. It addresses the issue at hand, so merging this.
